### DISCUSSION TITLE
How to deal with IBAN collisions

### DISCUSSION THREAD

**Discussion by DerRockWolf** [firefly-iii/firefly-iii#6106] [Firefly III]
It seems like some banks are reusing old IBANs for new accounts.
This is likely very uncommon, but it happened to once.

Example csv export:
date | opposing name | opposing IBAN
-|-|-
XX.08.2021 | Company A | XXXXXXXXXXXXXXXXXXXX05
XX.11.2021 | Company B | XXXXXXXXXXXXXXXXXXXX06
XX.05.2022 | Company B | XXXXXXXXXXXXXXXXXXXX05 (same as row1)

How would I deal with a case like this?
Would I just map the `opposing name` of the last transaction to the account name `Company B`, remove the IBAN of account `Company A` and update the IBAN of account `Company B` to reflect the new one? 

Is there a way to dry-run FIDI so that I know to which account FIDI would map the transaction to?
That would really help and make it obvious with detecting such cases. Because the opposing name of the csv would be different from the opposing account of the transaction that would be created.
It would be best if this dry-run would somehow be visible in the map data section. To directly see what FIDI would do (create a new account, map it to a wrong account, etc.).

**Comment by JC5** [comment]
I’ve never run into a recycled iban case, especially not on such a small timescale (ie 1 year).

But yes, I would remove the IBAN from the old Firefly III account and add it to the new one.
